### DISCUSSION TITLE
装了 nvm-desktop 后，对于原来的 nvm 安装的 node 有什么影响

### DISCUSSION THREAD

**Discussion by xmasuhai** [1111mp/nvm-desktop#42] [Q&A]
以前装的 pnpm yarn npm 会受影响吗

**Comment by 1111mp** [comment]
如果在系统中的环境变量`nvm-desktop`的优先级要比`nvm`高，那么是会有影响的（反之亦然），简单就是谁的优先级高，系统就是优先识别到并使用谁。

不过当你使用`nvm-desktop`的时候没有安装过`yarn`&`pnpm`，那么即使`nvm-desktop`的优先级高，如果这时候你在`nvm`中安装过，那么会识别到`nvm`安装的`yarn`&`pnpm`，是因为`nvm-desktop`在未安装`yarn`&`pnpm` 的情况下并不会添加其对应的垫片（用于拦截），所以系统会根据环境变量继续查找计算机中存在的`yarn`&`pnpm`的可执行文件，刚好`nvm`中有，那么就识别到并使用了。
